azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -91,10 +92,12 @@
   private final Props azkabanProps;
   private final Map<String, Props> sharedProps = new HashMap<>();
   private final JobRunnerEventListener listener = new JobRunnerEventListener();
+  private final FlowRunnerEventListener flowListener = new FlowRunnerEventListener();
```

Comment:
Consider getting the listener instance from Guice. It would be great to refactor FlowRunner to use Guice. For now getting the dependency from azkaban.ServiceProvider is fine.

Reply:
Ok, I will work on that in a separate PR.

Reply:
This refactoring is tricky. May have to introduce additional classes to hold state and the stateless part of the class can be injected via Guice.